File: Editorial/77_Combinations/Java/1_Approach_Backtracking.java
public void backtrack(
    List<Integer> curr,
    int firstNum,
    List<List<Integer>> ans
) {
    if (curr.size() == k) {
        ans.add(new ArrayList<>(curr));
        return;
    }

    for (int num = firstNum; num <= n; num++) {
        curr.add(num);
        backtrack(curr, num + 1, ans);
        curr.remove(curr.size() - 1);
    }

    return;
}

File: Editorial/77_Combinations/CPP/1_Approach_Backtracking.cpp
void backtrack(vector<int>& curr, int firstNum, vector<vector<int>>& ans) {
    if (curr.size() == k) {
        ans.push_back(curr);
        return;
    }

    for (int num = firstNum; num <= n; num++) {
        curr.push_back(num);
        backtrack(curr, num + 1, ans);
        curr.pop_back();
    }

    return;
}
